mohanadarafe/GuideMe
# Feature/detox setup

## **Contents of the pull request** 

In this PR, I have integrated **Detox**, a gray box end-to-end testing and automation library for mobile apps into the project.

### A basic test scenario has been added:

1. Load the app
2. Display main screen
3. Click on the searchbar
4. Type the query: "hall building"
 5. Select the corresponding item from the dropdown list
6. Click on the bottom menu (not yet implemented)
7. Scroll down and read the information provided (not yet implemented)

## **How to test:** 

**_Note:_** Testing this PR requires an iOS simulator.

1. checkout the branch feature/detox-setup:
`git checkout feature/detox-setup`

2. Run the following command:
`sh detox-setup.sh`

3. Open a separate terminal window and start the app:
`expo start`

4.  Go back to the previous terminal window and run the following command:
`npm run e2e`

5. The iOS simulator should show the previously mentioned test scenario

This PR fixes issues #160 and #131
